src/lib/marlin_plonk_bindings/stubs/src/caml_pointer.rs

```diff
@@ -5,14 +5,14 @@ use std::rc::Rc;
 pub struct CamlPointer<T>(pub Rc<T>);
 
 impl<T> CamlPointer<T> {
-    extern "C" fn caml_pointer_finalize(v: ocaml::Value) {
-        let v: ocaml::Pointer<CamlPointer<T>> = ocaml::FromValue::from_value(v);
+    extern "C" fn caml_pointer_finalize(v: ocaml::Raw) {
         unsafe {
+            let v: ocaml::Pointer<CamlPointer<T>> = v.as_pointer();
             v.drop_in_place();
         }
     }
 
-    extern "C" fn caml_pointer_compare(_: ocaml::Value, _: ocaml::Value) -> i32 {
+    extern "C" fn caml_pointer_compare(_: ocaml::Raw, _: ocaml::Raw) -> i32 {
```

Comment:
two ideas:

* why not just remove the `compare` field in `custom!` below?
* or why not panicking to make sure this is not used?